data/src/test/java/org/apache/iceberg/io/TestWriterMetrics.java

```diff
@@ -287,10 +287,10 @@ public void testMaxColumnsWithDefaultOverride() throws IOException {
         OutputFileFactory.builderFor(maxColumnTable, 1, 1).format(fileFormat).build();
 
     T row = toGenericRow(1, numColumns);
-    DataWriter dataWriter =
+    DataWriter<T> dataWriter =
```

Comment:
Unrelated note: I'm a big fan of the try-with-resource pattern, instead of calling the `.close()` (or forgetting to do so).

Reply:
Agree, as well, but I would make that a separate change since this behavior was preexisting.

Reply:
I think the reason it is written like this is that `DataFile dataFile = dataWriter.toDataFile();` below has to be called when dataWriter is closed.